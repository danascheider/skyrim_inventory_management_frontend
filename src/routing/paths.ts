export type PathString = `/${string}`

interface Paths {
  [i: string]: PathString
}

const paths: Paths = {
  home: '/',
  login: '/login'
}

export default paths
