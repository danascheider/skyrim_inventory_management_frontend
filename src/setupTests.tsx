import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

export const renderWithRouter = (ui: ReactElement) => render(<BrowserRouter>{ui}</BrowserRouter>)