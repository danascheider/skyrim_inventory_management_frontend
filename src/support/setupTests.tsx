import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import { LoginContext } from '../contexts/loginContext'
import { User } from 'firebase/auth'
import testProfileImg from './testProfileImg.png'

export const requireLogin = () => { /* noop */ }

export const testUser = {
  uid: 'edna',
  displayName: 'Edna St. Vincent Millay',
  email: 'edna@gmail.com',
  photoURL: testProfileImg,
  emailVerified: true,
  getIdToken: () => new Promise<string>((resolve, _reject) => resolve('xxxxxxx'))
} as User

export const renderWithRouter = (ui: ReactElement) =>
  render(<BrowserRouter>{ui}</BrowserRouter>)

export const renderAuthenticated = (
  ui: ReactElement,
  authLoading = false,
  authError = undefined
) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: testUser, token: 'xxxxxxx', requireLogin, authLoading, authError }}>
      {ui}
    </LoginContext.Provider>
  )

export const renderUnauthenticated = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: null, token: null, authLoading: false, requireLogin }}>
      {ui}
    </LoginContext.Provider>
  )

export const renderAuthLoading = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: null, token: null, authLoading: true, requireLogin }}>
      {ui}
    </LoginContext.Provider>
  )
