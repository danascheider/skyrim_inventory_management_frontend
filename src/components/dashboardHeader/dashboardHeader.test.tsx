import { describe, test, expect } from 'vitest'
import { renderWithRouter } from '../../setupTests'
import paths from '../../routing/paths'
import DashboardHeader from './dashboardHeader'

describe('<DashboardHeader />', () => {
  test('Dashboard header contains a link to the main dashboard', () => {
    const wrapper = renderWithRouter(<DashboardHeader />)
    expect(wrapper).toBeTruthy()

    const a = wrapper.container.querySelector('a')
    expect(a?.href).toBe(paths.dashboard.main)
  })

  test('matches snapshot', () => {
    const wrapper = renderWithRouter(<DashboardHeader />)
    expect(wrapper).toMatchSnapshot()
  })
})
