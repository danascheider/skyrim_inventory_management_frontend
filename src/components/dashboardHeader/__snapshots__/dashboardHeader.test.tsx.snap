// Vitest Snapshot v1

exports[`<DashboardHeader /> > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_root_c29812"
      >
        <div
          class="_bar_c29812"
        >
          <span
            class="_headerContainer_c29812"
          >
            <h1
              class="_header_c29812"
            >
              <a
                class="_headerLinkLarge_c29812"
                href="/dashboard"
              >
                Skyrim Inventory
                <br
                  class="_bp_c29812"
                />
                 Management
              </a>
              <a
                class="_headerLinkSmall_c29812"
                href="/dashboard"
              >
                S. I. M.
              </a>
            </h1>
          </span>
          <span
            class="_root_d263fb"
          >
            <div
              class="_main_d263fb"
            >
              <div
                class="_button_d263fb"
                role="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-bars _hamburger_d263fb"
                  data-icon="bars"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="_info_d263fb"
                >
                  <p
                    class="_name_d263fb"
                  >
                    Edna St. Vincent Millay
                  </p>
                  <p
                    class="_email_d263fb"
                  >
                    edna@gmail.com
                  </p>
                </span>
                <img
                  alt="User profile image"
                  class="_img_d263fb"
                  src="/src/testProfileImg.png"
                />
              </div>
            </div>
            <menu
              class="_dropdown_d263fb"
            >
              <div>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-right-from-bracket "
                  data-icon="right-from-bracket"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                    fill="currentColor"
                  />
                </svg>
                Sign Out
              </div>
            </menu>
          </span>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_root_c29812"
    >
      <div
        class="_bar_c29812"
      >
        <span
          class="_headerContainer_c29812"
        >
          <h1
            class="_header_c29812"
          >
            <a
              class="_headerLinkLarge_c29812"
              href="/dashboard"
            >
              Skyrim Inventory
              <br
                class="_bp_c29812"
              />
               Management
            </a>
            <a
              class="_headerLinkSmall_c29812"
              href="/dashboard"
            >
              S. I. M.
            </a>
          </h1>
        </span>
        <span
          class="_root_d263fb"
        >
          <div
            class="_main_d263fb"
          >
            <div
              class="_button_d263fb"
              role="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-bars _hamburger_d263fb"
                data-icon="bars"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="_info_d263fb"
              >
                <p
                  class="_name_d263fb"
                >
                  Edna St. Vincent Millay
                </p>
                <p
                  class="_email_d263fb"
                >
                  edna@gmail.com
                </p>
              </span>
              <img
                alt="User profile image"
                class="_img_d263fb"
                src="/src/testProfileImg.png"
              />
            </div>
          </div>
          <menu
            class="_dropdown_d263fb"
          >
            <div>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-right-from-bracket "
                data-icon="right-from-bracket"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                  fill="currentColor"
                />
              </svg>
              Sign Out
            </div>
          </menu>
        </span>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
